core/templates/domain/utilities/language-util.service.spec.ts

```diff
@@ -181,3 +182,36 @@ describe('Language util service', function() {
     }]);
   });
 });
+
+describe('Language Constants', () => {
+  it('should have no directional discrepancies between Content and Audio ' +
+  'languages', () => {
+    CONSTANTS.SUPPORTED_CONTENT_LANGUAGES.forEach((contentLanguage) => {
+      const matchingLanguage = CONSTANTS.SUPPORTED_AUDIO_LANGUAGES.find(
+        (audioLanguage) => audioLanguage.id === contentLanguage.code);
+      if (matchingLanguage !== undefined) {
+        expect(matchingLanguage.direction).toEqual(contentLanguage.direction);
+      }
```

Comment:
I think this would also create a need for a test to make sure that all language codes in the content and audio language constants exists in the separate map. Is there a better way to test this than iterating?

I agree though that the issue is in the overlap. Perhaps it would be better to have one constant with `isSpoken` and `isWritten` properties on them. This would make it clear if a language is spoken only or written only?

I'm not sure if such a fix should be included in this PR though, is it worth creating an issue?

Reply:
It might be worth the tradeoff in this case to iterate to make sure that all language codes in the content and audio language constants exist in the separate map. You would at least minimize the logic to one loop versus using a loop and a conditional statement here. You might be better off just having a separate test case for each language you want to test though.

Having one language list might be better long-term, though yes I agree this would be a lot more refactoring and probably better in a separate PR.

How about we go with the direction language map here and create an issue to refactor the language constants into one later? We don't want to introduce tests like this if possible and the additional language direction map changes would be minimal.